crates/context_aware_config/src/api/dimension/handlers.rs

```diff
@@ -86,39 +93,155 @@ async fn create(
-            e,
-        )) => {
-            log::error!("{fun_name:?} function not found with error: {e:?}");
-            Err(bad_argument!(
-                "Function {} doesn't exists",
-                fun_name.unwrap_or(String::new())
-            ))
+    })
+}
+
+#[put("/{name}")]
+async fn update(
+    path: Path<DimensionName>,
+    state: Data<AppState>,
+    req: web::Json<UpdateReq>,
+    user: User,
+    db_conn: DbConnection,
+    tenant_config: TenantConfig,
+) -> superposition::Result<HttpResponse> {
+    let name: String = path.clone().into();
+    use dimensions::dsl;
+    let DbConnection(mut conn) = db_conn;
+
+    let mut dimension_row: Dimension = dsl::dimensions
+        .filter(dimensions::dimension.eq(name.clone()))
+        .get_result::<Dimension>(&mut conn)?;
+
+    let update_req = req.into_inner();
+
+    if let Some(schema_value) = update_req.schema {
+        validate_jsonschema(&state.meta_schema, &schema_value)?;
+
+        let schema_compile_result = JSONSchema::options()
+            .with_draft(Draft::Draft7)
+            .compile(&schema_value);
+
+        if let Err(e) = schema_compile_result {
+            return Err(bad_argument!(
+                "Invalid JSON schema (failed to compile): {:?}",
+                e
+            ));
+        };
+        dimension_row.schema = schema_value;
+    }
+
+    let fun_name = match update_req.function_name {
+        Some(Value::String(func_name)) => Some(func_name),
+        Some(Value::Null) | None => None,
+        _ => {
+            log::error!("Expected a string or null as the function name.");
+            return Err(bad_argument!(
+                "Expected a string or null as the function name."
+            ));
         }
-        Err(e) => {
-            log::error!("Dimension upsert failed with error: {e}");
-            Err(unexpected_error!(
-                "Something went wrong, failed to create/update dimension"
+    };
+    dimension_row.function_name = fun_name.clone();
+
+    if let Some(position_val) = update_req.position.clone() {
+        let new_position: i32 = position_val.clone().into();
+        validate_dimension_position(path.into_inner(), position_val)?;
+
+        let previous_position = dimension_row.position.clone();
+        conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
+            if previous_position < new_position {
+                diesel::update(dsl::dimensions)
+                    .filter(dimensions::position.gt(dimension_row.position))
+                    .filter(dimensions::position.le(new_position.clone()))
+                    .set((
+                        dsl::last_modified_at.eq(Utc::now().naive_utc()),
+                        dsl::last_modified_by.eq(user.get_email()),
+                        dimensions::position.eq(dimensions::position - 1),
+                    ))
+                    .execute(transaction_conn)?
+            } else {
```

Comment:
if previous_position == new_position then we don't need to execute any position update query
```suggestion
            } else if  previous_position > new_position {
```